### PR TITLE
Improve alias remapping

### PR DESCRIPTION
**Pull request by TobiasFella**
- Don't print a warning if an alias is remapped from a room to its replacement room
- Don't remap aliases from the new room to the old room
(I've never noticed any problems caused by this, but the log message is slightly annoying)

### CODE REVIEW

**Review by KitsuneRal — APPROVED** [review]

**Comment by TobiasFella** [comment]
Sonar complains about "code complexity"; should i refactor the code a bit?

**Comment by KitsuneRal** [comment]
TL;DR: No; it's Sonar's fault, not ours.

I found some time ago that Sonar spins up complexity on every `qC*` macro (you can see it in the issue details). I used to switch to their `q*` counterparts in the past to avoid it (they also allow to specify logging categories since Qt 5.12 or something). I don't practice it any more because, as I mentioned in the chat one day, `qC*` versions force you to provide a category (a good thing) while `q*` don't; and this is more valuable than trying to go around quirks of a static analysis tool.

And after digging up stuff recently I learnt that Sonar actually [tried to fix this very issue](https://github.com/SonarOpenCommunity/sonar-cxx/pull/1552) as long ago as in 2018 - yet somehow it's still here. Looking at the way it was fixed, I can imagine it still tries to account for "generated code" (as it's called inside Sonar codebase) in some place - so, see TL:DR.
